cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -644,28 +647,30 @@ public void getProgramStatusSchedules(HttpRequest request, HttpResponder respond
       : new ApplicationId(triggerNamespaceId, triggerAppName, triggerAppVersion);
     final ProgramId triggerProgramId = appId.program(ProgramType.valueOfCategoryName(triggerProgramType),
                                                      triggerProgramName);
-    Collection<ProgramScheduleRecord> schedules = programScheduler.findSchedules(triggerProgramId.toString());
-    if (triggerProgramStatuses != null) {
-      final Set<co.cask.cdap.api.ProgramStatus> queryProgramStatuses = new HashSet<>();
-      try {
-        for (String status : triggerProgramStatuses.split(",")) {
+
+    Collection<ProgramScheduleRecord> schedules = new HashSet<>();
+    Set<co.cask.cdap.api.ProgramStatus> queryProgramStatuses = new HashSet<>();
+
+    if (triggerProgramStatuses == null) {
+      queryProgramStatuses = ImmutableSet.of(co.cask.cdap.api.ProgramStatus.COMPLETED,
```

Comment:
Yeah, might be better if this was also in the enum itself? I know that we have a `isDone` method in the `ProgramController.State` enum that is used similarly.